docs/docs/tutorials/6-tools.md

```diff
@@ -121,29 +121,82 @@ Please note that tools/function calling is not the same as [JSON mode](/tutorial
 
 ## 2 levels of abstraction
 
-LangChain4j provides two levels of abstraction for working with tools.
+LangChain4j provides two levels of abstraction for using tools.
 
 ### Low level Tool API
-At the low level, you can use the `generate(List<ChatMessage>, List<ToolSpecification>)`
-and `generate(List<ChatMessage>, ToolSpecification)` methods
-of `ChatLanguageModel` (and similar methods of `StreamingChatLanguageModel`).
 
-You'll need to manually create `ToolSpecification` object(s) containing all information about the tool,
-or use the `ToolSpecifications.toolSpecificationFrom(Method)` helper method
-to convert any Java method into a `ToolSpecification`.
+At the low level, you can use the `generate(List<ChatMessage>, List<ToolSpecification>)` method
+of the `ChatLanguageModel`. A similar method is also present in the `StreamingChatLanguageModel`.
 
-When the LLM decides to call the tool, the returned `AiMessage` will have data
-in a `List<ToolExecutionRequest> toolExecutionRequests` field instead of a `String text` field.
-Depending on the LLM, it can contain one or multiple `ToolExecutionRequest`s
+`ToolSpecification` is an object that contains all the information about the tool:
+- The `name` of the tool
+- The `description` of the tool
+- The `parameters` of the tool
+
+It is recommended to provide as much information about the tool as possible:
+a clear name, a complete description, and a description for each parameter, etc.
+
+There are two ways to create a `ToolSpecification`:
+
+1. Manually
+```java
+ToolSpecification toolSpecification = ToolSpecification.builder()
+    .name("getWeather")
+    .description("Returns the weather forecast for a given city")
+    .addParameter("city", type("string"), description("The city for which the weather forecast should be returned"))
+    .addParameter("temperatureUnit", enums(TemperatureUnit.class)) // enum TemperatureUnit { CELSIUS, FAHRENHEIT }
+    .build();
+```
+
+2. Using helper methods:
+- `ToolSpecifications.toolSpecificationsFrom(Class)`
+- `ToolSpecifications.toolSpecificationsFrom(Object)`
+- `ToolSpecifications.toolSpecificationFrom(Method)`
+
+```java
+class WeatherTools { 
+  
+    @Tool("Returns the weather forecast for a given city")
+    String getWeather(
+            @P("The city for which the weather forecast should be returned") String city,
+            TemperatureUnit temperatureUnit
```

Comment:
We can parse Enum, but nested object parameters are not supported yet. Is it correct?

Reply:
Yes, enums are supported: https://github.com/langchain4j/langchain4j/blob/1827302342eee393fddfce8a07f3cab2b7d5b9e2/langchain4j-core/src/test/java/dev/langchain4j/agent/tool/ToolSpecificationsTest.java#L60

Nested POJOs will also be supported soon: https://github.com/langchain4j/langchain4j/pull/708